### ISSUE TITLE
Cannot put Tooltip on  OverflowMenu custom trigger icon

### ISSUE THREAD

**Issue by bitfede** [carbon-design-system/carbon#7279]
## What package(s) are you using?

- [ ] `carbon-components`
- [x ] `carbon-components-react`

## Detailed description

> Describe in detail the issue you're having.

I cannot put a Tooltip on an OverflowMenu icon.

> Is this issue related to a specific component?

OverflowMenu and TooltipIcon

> What did you expect to happen? What happened instead? What would you like to
> see changed?

I expected this to happen without errors:
<img width="339" alt="Screen Shot 2020-11-12 at 1 51 56 AM" src="https://user-images.githubusercontent.com/15684622/98906284-cc809e00-248a-11eb-9275-0f0cce21e261.png">

Instead, I got this error in the console:
```
Warning: validateDOMNesting(...): <button> cannot appear as a descendant of <button>.
```
I would love to have the ability to put a Tooltip on an OverflowMenu button.

> What browser are you working in?

Chrome

>My Code

```
<OverflowMenu
    style={{height: '3rem', maxHeight: '3rem', width: '3rem', background: '#f4f4f4'}}
    renderIcon={() => <TooltipIcon direction={"top"} align={"center"} style={{borderBottom: '0'}} tooltipText={"Status"}> <EventSchedule20 /> </TooltipIcon>}
>
{menuItems}
</OverflowMenu>
```

Thank you very much for your time! Hopefully there's a workaround or a solution I'm missing.

**Comment by bitfede** [comment]
Tried also with 
```
      <OverflowMenu
        style={{height: '3rem', maxHeight: '3rem', width: '3rem', background: '#f4f4f4'}}
        renderIcon={() => <Button renderIcon={Renew20} iconDescription="Refresh" } hasIconOnly />}
        >
        {menuItems}
      </OverflowMenu>
```

But the problem still persists, a button ends up being rendered inside another button

**Comment by bitfede** [comment]
Tried again with:
```
      <OverflowMenu
        style={{height: '3rem', maxHeight: '3rem', width: '3rem', background: '#f4f4f4'}}
        renderIcon={() => <TooltipDefinition direction={"top"} align={"center"} style={{borderBottom: '0'}} tooltipText={"Status"}><EventSchedule20 />  </TooltipDefinition>}
        >
        {menuItems}
      </OverflowMenu>
```

But got this error:
```
index.js:1 Warning: Failed prop type: Invalid prop `children` of type `array` supplied to `TooltipDefinition`, expected `string`.
```
Because only a string is accepted as children value for TooltipDefinition.

**Comment by emyarod** [comment]
I'm not sure if this pattern is supported but as a workaround you can try something like this

```jsx
<OverflowMenu
  icon={yourCustomIconData}
  className="bx--tooltip__trigger bx--tooltip--icon__bottom"
  aria-label="your tooltip label"
>
  {menuItems}
</OverflowMenu>
```

**Comment by bitfede** [comment]
Hey, thank you so much for your tip, @emyarod . It seems like I'm getting closer to the correct implementation with this workaround.

<img width="194" alt="Screen Shot 2020-11-13 at 8 09 16 PM" src="https://user-images.githubusercontent.com/15684622/99134533-2351b900-25ec-11eb-8683-a0492602151e.png">

maybe I need to play around a bit more with the alignments. Any relevant piece of documentation / source code I can look into, other than the OverflowMenu.js source code?

Also where does the tooltip text "Menu" comes from?

Thank you very much! 👍

**Comment by emyarod** [comment]
you can add the `.bx--tooltip--align-start` (or `align-end`) class to change the alignment. these classes are just taken from the definition and icon tooltip

I'm not entirely sure where the text comes from without seeing your source code but it most likely is the text within `.bx--assistive-text` if you are following the tooltip/icon-only button markup

**Comment by bitfede** [comment]
Thank you very much for your feedback @emyarod ! I managed to get it done! Here it is 😃  :

<img width="356" alt="Screen Shot 2020-11-20 at 8 34 36 PM" src="https://user-images.githubusercontent.com/15684622/99864211-1e52b380-2b70-11eb-81ab-efce28cec56d.png">

<img width="242" alt="Screen Shot 2020-11-20 at 8 34 49 PM" src="https://user-images.githubusercontent.com/15684622/99864214-21e63a80-2b70-11eb-884c-cd88749f54e8.png">

To achieve this I added the following props:

```
      <OverflowMenu
        ...
        className="bx--tooltip__trigger bx--tooltip--icon__top leftDropDownBtn"
        ariaLabel={"Category"}
        >
        {menuItems}
      </OverflowMenu>
```
And this CSS rule applied to these selectors:

```
.leftDropDownBtn.bx--tooltip--icon__top::after {
    transform: translate(20%, calc(-100% - 12px));
}
```
(previously the first argument to translate was -50%, that's why it was going over the endge of the page)

So the reason why it was rendering `"Menu"` instead of `"your tooltip label"` was because I was using the prop `aria-label` instead of `ariaLabel` in camelcase. As soon as I renamed the prop from `aria-label` to `ariaLabel` the correct string showed up.

I'll close the issue, hopefully anyone that has the same problem can find this helpful.
